### ISSUE TITLE
[Feature Request] Paypal Fee

### ISSUE THREAD

**Issue by flippz** [FOSSBilling/FOSSBilling#500]
Would be nice if we could get the amount from Paypal with the Paypal fee withdrawn so to see what you actually get.

**Comment by notbasetwo** [comment]
I think that's a nice idea, but would require much tighet integration with PayPal's API than what we have currently.

May be something for a long way down the road.

**Comment by BelleNottelling** [comment]
> I think that's a nice idea, but would require much tighet integration with PayPal's API than what we have currently.
> 
> May be something for a long way down the road.

Actually, it would probably be a relatively small change. The FOSSBilling end would need to be updated somewhere to create a config option for it, but in the IPN data there is `mc_fee` which represents the transaction fee. (`mc_gross` minus `mc_fee` equals the total amount deposited int onto the receiver account)

https://developer.paypal.com/api/nvp-soap/ipn/IPNandPDTVariables/

**Comment by adamh2021** [comment]
The ability to add this to all payment gateways would be beneficial. I've seen it before on other software - the ability to set a % and fixed fee per transaction gateway:

Not something that's linked to the gateway, something that you can manually configure within the software package. 
I consider this rather an important feature - PayPal are charging 1.2% + £0.30 per transaction at the moment, really could do with passing that on to customers!

**Comment by notbasetwo** [comment]
Whilst off topic on a technical perspective.

I would warn all that in most territories, surcharging (adding a fee on) for the use of PayPal is prohibited in the PayPal User Agreement and could see your account with them terminated.

**Comment by adamh2021** [comment]
Apologies for being off topic - the feature request is what attracted my attention. 

In the UK - We are allowed to charge transaction fee's for Business to Business transactions, but the fee must not exceed cost directly borne by them. 

For Business to consumer fees - we are not allowed to add a "Transaction Fee". But we are allowed to add an 'administration fee".  With regards to PayPal allowing this to take place - I was not aware of their rule and I am checking with PayPal now. If that is the case, Stripe do allow this. 

Either way - It would be nice feature to have and it will be the responsibility of the FossBilling user to ensure that they are using this feature lawfully, not the software developers.

**Comment by admdly** [comment]
> In the UK - We are allowed to charge transaction fee's for Business to Business transactions, but the fee must not exceed cost directly borne by them. 

This is correct - you can pass along any fees for a particular transaction for B2B transactions under UK law. 

> For Business to consumer fees - we are not allowed to add a "Transaction Fee". But we are allowed to add an 'administration fee".  With regards to PayPal allowing this to take place - I was not aware of their rule and I am checking with PayPal now. If that is the case, Stripe do allow this. 

The UK law forbids adding a surcharge on the basis of the consumer's chosen payment method. That is you cannot add a charge simply because they choose to pay a certain way (regardless of what you call the fee). 

Stripe and PayPal may allow adding such charges, but to do so would be illegal in the UK. 


> Either way - It would be nice feature to have and it will be the responsibility of the FossBilling user to ensure that they are using this feature lawfully, not the software developers. 

Agreed - it would be virtually impossible to keep up with the various laws globally as many have subtle difference. For example, the UK law includes payment service providers like PayPal, whereas I believe the EU regulation does not.
